docs/index.md

```diff
@@ -4,396 +4,182 @@
-maintained by the Genkit team:
-
-<table class="responsive">
-  <tr>
-    <th colspan=2>Official plugins</th>
-  </tr>
-  <tr>
-    <td><a href="./plugins/google-genai.md"><code>googleai</code></a></td>
-    <td>
-      <b>Generative models</b>: Gemini Pro, Gemini 1.5 Pro, Gemini Pro Vision<br>
-      <b>Embedding models</b>: Gecko text embedding<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/vertex-ai.md"><code>vertexai</code></a></td>
-    <td>
-      <b>Generative models</b>: Gemini Pro, Gemini Pro Vision, Gemini 1.5 Flash, Gemini 1.5 Pro, Imagen2, Anthropic Claude 3<br>
-      <b>Embedding models</b>: Gecko text embedding<br>
-      <b>Evaluators</b>: Vertex AI evaluation
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/ollama.md"><code>ollama</code></a></td>
-    <td>
-      <b>Generative models</b>: Many local models, including Gemma, Llama 3, Mistral, and more<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/chroma.md"><code>chroma</code></a></td>
-    <td>
-      <b>Vector database</b>: ChromaDB<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/pinecone.md"><code>pinecone</code></a></td>
-    <td>
-      <b>Vector database</b>: Pinecone<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/google-cloud.md"><code>google-cloud</code></a></td>
-    <td>
-      <b>Monitoring tools</b>: Google Cloud Trace, Google Cloud Logging<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/firebase.md"><code>firebase</code></a></td>
-    <td>
-      <b>Cloud deployment</b>: Cloud Functions, Firebase Authentication, App Check<br>
-      <b>Vector database</b>: Cloud Firestore vector store<br>
-    </td>
-  </tr>
-  <tr>
-    <td><a href="./plugins/langchain.md"><code>langchain</code></a></td>
-    <td>
-      Use LangChain chains and utilities in Genkit flows
-    </td>
-  </tr>
-</table>
+  });
+
+  const result = await generate({
+      model: gemini15Flash,
+      tools: [lookupWeather], // Give the model a list of tools it can call
+      prompt: 'What is the weather like in New York? ',
+  });
+  ```
+
+* {Retrieval}
+
+  ```javascript
+  import { generate, retrieve } from `@genkit-ai/ai`;
+  import { devLocalRetrieverRef } from '@genkit-ai/dev-local-vectorstore';
+  import { gemini15Flash } from `@genkit-ai/googleai`;
+
+  // Reference to a local vector database storing Genkit documentation
+  const retriever = devLocalRetrieverRef('genkitQA');
+
+  const query = 'How do I retrieve relevant documents in Genkit?'
+
+  // Consistent API to retrieve documents
+  const docs = await retrieve({
+      retriever: retriever,
+      query: query,
+      options: { limit: 5 },
+  });
+
+  const result = await generate({
+      model: gemini15Flash
+      prompt: 'Use the provided context from the Genkit documentation to answer the following query: ${query}',
+      context: docs // Pass retrieved documents to the model
+  });
+  ```
+
+* {Prompt template}
+
```

Comment:
Describe the advantages of prompt templates here. 

A simple but powerful standard for describing prompts in a typesafe way.  All the data needed to generate an LLM output in a compact format that works great as code, and with the Genkit developer tools

Reply:
Does that explanation work for both code-based dotprompts and .prompt files?

Reply:
I'm keeping the high level value prop in the table above. These are really only meant to be illustrative examples of what I talked about in the table.

I want to, as much as possible, drive folks to the Get started page.